client/components/App.jsx

```diff
@@ -58,10 +16,16 @@ class App extends React.Component {
   constructor() {
     super();
     this.state = {
-      allReviews: {},
+      reviewData: [],
```

Comment:
For `App`'s state, it does seem better to only store the reviews for the current product, and not all reviews for all products as it was before, since this is unnecessary info needed.